client/components/domains/registrant-extra-info/fr-form.jsx

```diff
@@ -0,0 +1,301 @@
+import debugFactory from 'debug';
+
+/**
+ * Internal dependencies
+ */
+import FormFieldset from 'components/forms/form-fieldset';
+import FormLabel from 'components/forms/form-label';
+import FormLegend from 'components/forms/form-legend';
+import FormRadio from 'components/forms/form-radio';
+import FormCountrySelect from 'components/forms/form-country-select';
+import FormTextInput from 'components/forms/form-text-input';
+import FormSettingExplanation from 'components/forms/form-setting-explanation';
+import { forDomainRegistrations as countriesListForDomainRegistrations } from 'lib/countries-list';
+
+const debug = debugFactory( 'calypso:domains:registrant-extra-info' );
+
+const countriesList = countriesListForDomainRegistrations();
+
+class RegistrantExtraInfoForm extends React.PureComponent {
+	static propTypes = {
+		isVisible: PropTypes.bool,
+		onSubmit: PropTypes.func,
+		countriesList: PropTypes.object,
+		onStateChanged: PropTypes.func, // Just until we can reduxify the contact details
+	}
+
+	static defaultProps = {
+		isVisible: true,
+		isProbablyOrganization: false,
+		onSubmit: noop,
+		countriesList: countriesList,
+	}
+
+	constructor( props ) {
+		super( props );
+	}
+
+	componentWillMount() {
+		this.setState( {
+			registrantType: this.props.isProbablyOrganization
+				? 'organization' : 'individual',
+		} );
+	}
+
+	componentDidMount() {
+		// Make sure that we send an an initial update in case the user submits
+		// without changing anything.
+		this.props.onStateChange &&
+			this.props.onStateChange( this.state );
+	}
+
+	getRelevantFields( state ) {
+		const { countryOfBirth, registrantType } = state;
+		const bornInFrance = countryOfBirth === 'FR';
+		const birthPlaceFields = [ 'placeOfBirth', 'postalCodeOfBirth' ];
+		const individualFields = [
+			'countryOfBirth',
+			'dateOfBirth',
+			...( bornInFrance ? birthPlaceFields : [] ),
+		];
+		const organizationalFields = [ 'registrantVatId', 'sirenSiret', 'trademarkNumber' ];
+		const conditionalFields = registrantType === 'individual'
+			? individualFields
+			: organizationalFields;
+		return [ 'registrantType', ...conditionalFields ];
+	}
+
+	handleDobChangeEvent = ( event ) => {
+		this.handleChangeEvent( event );
+		// setState() is not syncronous :(
+		const updatedState = {
+			...this.state,
+			[ event.target.name ]: event.target.value,
+		};
+		const { dob_years: years, dob_months: months, dob_days: days } = updatedState;
+
+		if ( years && months && days ) {
+			const value = [ years, months, days ].join( '-' );
+			const pseudoEvent = {
+				target: {
+					name: 'dateOfBirth',
+					value
+				}
+			};
+			debug( 'Setting dateOfBirth to ' + value +
+				( moment( value, 'YYYY-MM-DD' ).isValid() ? '' : ' (invalid)' ) );
+			this.handleChangeEvent( pseudoEvent );
+		}
+	}
+
+	handleChangeEvent = ( event ) => {
+		this.setState( { [ event.target.name ]: event.target.value } );
+	}
+
+	shouldComponentUpdate = ( nextProps, nextState ) => {
```

Comment:
Nearly, but not quite :( `isEqual()` is a deep comparison, and PureComponent uses a shallow one https://facebook.github.io/react/docs/react-api.html#react.purecomponent.

I had to add this to break an infinite rerender caused by passing the state up out of this component and adding it to the state of it's parent. This was the point at which I decided that I had forced this too far, and need to stop and reduxify it all :)